TAScheduler/UnitTests/tests.py

```diff
@@ -178,50 +189,778 @@ class TestCourseGetCrseInfo(TestCase):  # Randall
+
+    def test_get_parent_course(self):
+        self.assertEquals(self.lab.getParentCourse(), self.course,
+                          "getParentCourse() did not retrieve correct course")
 
 
 class TestLabInit(TestCase):
-    pass
+    lab = None
+    info = None
+    tempLab = None
+
+    # noinspection DuplicatedCode
+    def setUp(self):
+        tempuser = User(
+            email_address="test@test.com",
+            password="password",
+            first_name="first",
+            last_name="last",
+            home_address="Your mom's house",
+            phone_number=1234567890
+        )
+
+        tmpta = TA.objects.create(
+            user=tempuser,
+            grader_status=True
+        )
+
+        tmpcourse = Course.objects.create(
+            course_id=100,
+            semester="fall 2023",
+            name="testCourse",
+            description="test",
+            num_of_sections=3,
+            modality="online",
+            credits=3
+        )
+
+        tmpsection = Section.objects.create(
+            section_id=1011,
+            course=tmpcourse,
+            location="Cool place",
+            meeting_time=datetime.datetime
+        )
+
+        self.tmplab = Lab.objects.create(
+            section=tmpsection,
+            ta=tmpta
+        )
+
+        self.lab = LabObj(self.tmplab)
+
+    def test_lab_make(self):
+        self.assertIsNotNone(LabObj(self.tempLab), "__init__ failed in making Lab")
+
+    def test_bad_lab_make(self):
+        with self.assertRaises(TypeError, msg="__init__ failed in making Lab, bad input"):
+            LabObj(5)
+
 
 class TestLabGetLabTAAsgmt(TestCase):  # Joe
-    pass
+    lab = None
+    ta = None
+
+    # noinspection DuplicatedCode
+    def setUp(self):
+        tmpcourse = Course.objects.create(
+            course_id=100,
+            semester="fall 2023",
+            name="testCourse",
+            description="test",
+            num_of_sections=3,
+            modality="online",
+            credits=3
+        )
+
+        tmpsection = Section.objects.create(
+            section_id=1011,
+            course=tmpcourse,
+            location="Cool place",
+            meeting_time=datetime.datetime
+        )
+
+        self.lab = Lab.objects.create(
+            section=tmpsection
+        )
+        self.lab.save()
+
+    def test_get_ta(self):
+        temp = User(
+            email_address="test@test.com",
+            password="password",
+            first_name="first",
+            last_name="last",
+            home_address="Your mom's house",
+            phone_number=1234567890
+        )
+        temp.save()
+        self.ta = TA.objects.create(user=temp)
+        self.ta.save()
+
+        self.lab.ta = self.ta
+        self.lab.save()
+
+        self.labObj = LabObj(self.lab)  # Form lab after adding TA manually
+
+        self.assertEquals(self.ta, self.labObj.getLabTAAsgmt(), "getLabTAAsgmt() does not retrieve correct ta")
```

Comment:
Honestly not sure whether .getLabTAAsgmt should return the database TA or a TA object ... I would put a quick comment here to note that if something is going wrong, this might be why